From 1920d52d6f37b57d43f9adf533fd40045fe3132b Mon Sep 17 00:00:00 2001
From: delhiryder <sidd.gupta@gmail.com>
Date: Tue, 21 Nov 2023 01:41:09 +0000
Subject: [PATCH] Changed the behavior of the wlan_en (regulator-3) to
 'regulator-off-in-suspend'. Fix for the wlcore/wl1271_sdio_mmc2 errors.

---
 arch/arm64/boot/dts/ti/k3-am625-beagleplay.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am625-beagleplay.dts b/arch/arm64/boot/dts/ti/k3-am625-beagleplay.dts
index eda9b77f49ac..0f6ef0693105 100644
--- a/arch/arm64/boot/dts/ti/k3-am625-beagleplay.dts
+++ b/arch/arm64/boot/dts/ti/k3-am625-beagleplay.dts
@@ -140,7 +140,8 @@ wlan_en: regulator-3 {
 		regulator-min-microvolt = <1800000>;

 		regulator-max-microvolt = <1800000>;
 		enable-active-high;
-		regulator-always-on;
+		/* regulator-always-on; */
+		regulator-off-in-suspend;
 		vin-supply = <&vdd_3v3>;
 		gpio = <&main_gpio0 38 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
